### ISSUE TITLE
Unexpected assertion failed: self.primary_key.is_none()

### ISSUE THREAD

**Issue by andylokandy** [tikv/client-rust#225]
I modified examples/transaction.rs to the following, and then it panicked.

```rust
#[tokio::main]
async fn main() {
    let args = parse_args("txn");
    let client = Client::new(args.pd)
        .await
        .expect("Could not connect to tikv");

    let mut txn = client.begin_optimistic().await.unwrap();
    
    txn.put("k1".to_owned(), b"v2".to_owned()).await.unwrap();
    txn.scan(("k1".to_owned()).., 10).await.unwrap();
    
    txn.commit().await.unwrap();
}

```

```
> cargo run --example transaction -- --pd 127.0.0.1:2379
Finished dev [unoptimized + debuginfo] target(s) in 18.35s
     Running `target/debug/examples/transaction --pd '127.0.0.1:2379'`
thread 'main' panicked at 'assertion failed: self.primary_key.is_none()', /home/andylokandy/Code/client-rust/src/transaction/transaction.rs:684:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by ekexium** [comment]
Probably because of the scan overriding entries in the local buffer. It should be fixed in #218. Could you try that?

**Comment by longfangsong** [comment]
Aleady fixed in #218.
